src/ast/passes/semantic_analyser.cpp

```diff
@@ -2677,6 +2678,18 @@ void SemanticAnalyser::visit(AssignVarStatement &assignment)
     if (ty == Type::none)
       LOG(ERROR, assignment.expr->loc, err_)
           << "Invalid expression for assignment: " << ty;
+
+    // Scratch variables are on stack. The BPF stack is only 512 bytes at time
+    // of writing, so large values do not fit on there. 200 is a ballpark of
+    // what probably won't work.
+    auto expr_size = assignTy.GetSize();
+    if (expr_size > 200) {
+      LOG(ERROR, assignment.loc, err_)
+          << "Value is too big "
+          << "(" << expr_size << " bytes) for the stack. "
+          << "Try reducing its size, storing it in a map, or creating it in "
```

Comment:
Maybe a small example for the the last option wouldn't be bad.

Reply:
Actually, an example for the size reduction would be good to.

Reply:
Updated. I originally didn't wanna get too specific as multiple types can trigger this (tuples included). But probably ok to optimize for common case strings. It's hopefully obvious enough

Reply:
I like it! I always find examples super helpful.